### DISCUSSION TITLE
[REGRESSION] Running in a monorepo with different configurations

### DISCUSSION THREAD

**Discussion by orgads** [eslint/eslint#18841] [Q&A] [answered]
Hi,

I have a repo with `common` directory and some other services.

With eslint 8, I run `git ls-files \*.ts | xargs -r common/node_modules/.bin/eslint --ext .ts -f checkstyle > eslint.xml`, and eslint runs in all the directories, each with its own configuration.

With eslint 9 this doesn't work. I have to install eslint and all its dependencies in the root directory, and even if I do that, only the root settings will be used.

Can you suggest a way to achieve the same behavior with eslint 9? Is this a known issue? Was it intentional?

**Comment by nzakas** [comment, marked as answer]
Take a look at the experimental config lookup flag:
https://eslint.org/docs/latest/use/configure/configuration-files#experimental-configuration-file-resolution

**Reply by orgads** [reply]
It works! Just one problem I had with it - eslint could not find `checkstyle` formatter, although it was installed in all of the sub-projects (but not in the root directory).

**Reply by nzakas** [reply]
If you're running ESLint from the root directory then you'll need to install the formatter at the root directory.

**Reply by orgads** [reply]
But the root directory has no package.json, it is only the base for other projects.

**Reply by nzakas** [reply]
Then don't run ESLint from the root directory.

**Reply by orgads** [reply]
Why doesn't it use plugins from its relative path (in my example, I run `common/node_modules/.bin/eslint`, so I'd expect it to search for the plugins under `common/node_modules`)?

**Reply by nzakas** [reply]
That's just how Node.js works.

**Reply by orgads** [reply]
Ok, thanks for the information.
